zap/src/main/dist/zap.sh

```diff
@@ -30,6 +30,11 @@ if [ "$OS" = "Darwin" ]; then
   fi
 fi
 
+# On Linux, respect JAVA_HOME if it exists in the environment
+if [ "$OS" = "Linux" ] && [ -n "$JAVA_HOME" ]; then
```

Comment:
Could use `elif` as this wont be the case if its OS X

Reply:
Think I've corrected this with the latest commit.